client/src/pages/Search/index.js

```diff
@@ -76,19 +78,42 @@ export default function Search() {
   });
 
   const { longitude, latitude, meters, budget } = searchFields;
+  const pageNavigationProps = {
+    total_count: count,
+    page_number: page,
+    count_limit: 20,
+    updatePage: updatePage,
+  };
 
+  // Search on coordinate change
   useEffect(() => {
     const abortController = new AbortController();
-    retrieveRestaurants(searchFields, abortController);
+    setPage(0);
+    const query = { ...searchFields, page };
+    retrieveRestaurants(query, abortController);
 
-    localStorage.setItem("budget-eats-cache", JSON.stringify(searchFields));
+    localStorage.setItem("budget-eats-cache", JSON.stringify(query));
 
     return () => {
       abortController.abort();
     };
     // eslint-disable-next-line
   }, [longitude, latitude, meters, budget]);
 
+  // Search on page change
+  useEffect(() => {
+    const abortController = new AbortController();
+    const query = { ...searchFields, page };
+    retrieveRestaurants(query, abortController);
+
+    localStorage.setItem("budget-eats-cache", JSON.stringify(query));
+
+    return () => {
+      abortController.abort();
+    };
```

Comment:
Just so I'm sure, lines 111-113 (same with lines 97-99), it's always aborting any existing web requests whenever there's a change or another request occurring?

Reply:
Yep.